pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -390,23 +376,34 @@ KUBECTL=$(getversion kubectl ${KUBELET_VERSION}-)
 # Explicit cni version is a temporary workaround till the right version can be automatically detected correctly
 apt-get install -y kubelet=${KUBELET} kubeadm=${KUBEADM} kubectl=${KUBECTL}
 
+{{- end }} {{/* end install */}}
+
+{{ define "configure" -}}
+TOKEN={{ .Token }}
+MASTER={{ index .Cluster.Status.APIEndpoints 0 | endpoint }}
+MACHINE={{ .Machine.ObjectMeta.Namespace }}/{{ .Machine.ObjectMeta.Name }}
+NODE_LABEL_OPTION={{ if .Machine.Spec.Labels }}--node-labels={{ labelMap .Machine.Spec.Labels }}{{ end }}
+NODE_TAINTS_OPTION={{ if .Machine.Spec.Taints }}--register-with-taints={{ taintMap .Machine.Spec.Taints }}{{ end }}
+
+# Disable swap otherwise kubelet won't run
+swapoff -a
+sed -i '/ swap / s/^/#/' /etc/fstab
+
 systemctl enable docker || true
 systemctl start docker || true
 
 sysctl net.bridge.bridge-nf-call-iptables=1
 
-# kubeadm uses 10th IP as DNS server
-CLUSTER_DNS_SERVER=$(prips ${SERVICE_CIDR} | head -n 11 | tail -n 1)
```

Comment:
What are you doing in place of this?  You're removing this, but I don't see the replacement.  Actually, we already discussed it, but can you reply for historical tracking purposes?

Reply:
The reason this is not needed is that the kubeadm generates that value in the kubelet configuration already. Thus this is not needed anymore. I am not sure from which version of kubeadm this behavior changed, but with v 1.11.x and newer it seems to be the case.